src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -661,6 +661,21 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private void NewPackageSource_TextChanged(object sender, EventArgs e)
+        {
+            var source = NewPackageSource.Text.Trim();
+
+            // Warn if the source is http, support for this will be removed
+            if (!source.StartsWith("https", StringComparison.OrdinalIgnoreCase))
```

Comment:
Would local sources start with `file://` and make this warning visible?  Should we just check for `StartsWith("http:")` ?

Reply:
You can use the IsHttp and IsHttps on the PackageSource object.